### PR TITLE
Loki-stack: fix values for logstash and filebeat

### LINKED ISSUE #136: Wrong values for logstash and filebeat in loki-stack
Exemplary values in [*Loki-stack/values.yaml*](https://github.com/grafana/helm-charts/blob/main/charts/loki-stack/values.yaml) use probably some old version of logstash subchart and doesn't work.

### Filebeat 

Filebeat's `output.logstash.hosts` points to `["logstash-loki:5044"]` when in my case this service was named `loki-logstash-headless`. Corrected version:

```yaml
filebeat:
  enabled: true
  filebeatConfig:
    filebeat.yml: |-
      # logging.level: debug
      filebeat.inputs:
      - type: container
        paths:
          - /var/log/containers/*.log
        processors:
        - add_kubernetes_metadata:
            host: ${NODE_NAME}
            matchers:
            - logs_path:
                logs_path: "/var/log/containers/"
      output.logstash:
        hosts: ["loki-logstash-headless:5044"]
```

### Logstash

In this case format of *values.yaml* pulled with `helm pull grafana/loki-stack` logstash subchart is:

```yaml
---
replicas: 1

# Allows you to add any config files in /usr/share/logstash/config/
# such as logstash.yml and log4j2.properties
#
# Note that when overriding logstash.yml, `http.host: 0.0.0.0` should always be included
# to make default probes work.
logstashConfig: {}
#  logstash.yml: |
#    key:
#      nestedkey: value
#  log4j2.properties: |
#    key = value

# Allows you to add any pipeline files in /usr/share/logstash/pipeline/
### ***warn*** there is a hardcoded logstash.conf in the image, override it first
logstashPipeline: {}
#  logstash.conf: |
#    input {
#      exec {
#        command => "uptime"
#        interval => 30
#      }
#    }
#    output { stdout { } }

...
```

Which doesn't correspond with values in Loki-stack/values.yaml file which are:

```yaml
logstash:
  enabled: false
  image: grafana/logstash-output-loki
  imageTag: 1.0.1
  filters:
    main: |-
      filter {
        if [kubernetes] {
          mutate {
            add_field => {
              "container_name" => "%{[kubernetes][container][name]}"
              "namespace" => "%{[kubernetes][namespace]}"
              "pod" => "%{[kubernetes][pod][name]}"
            }
            replace => { "host" => "%{[kubernetes][node][name]}"}
          }
        }
        mutate {
          remove_field => ["tags"]
        }
      }
  outputs:
    main: |-
      output {
        loki {
          url => "http://loki:3100/loki/api/v1/push"
          #username => "test"
          #password => "test"
        }
        # stdout { codec => rubydebug }
      }
```

Correct version:

```yaml
logstash:
  enabled: true
  image: grafana/logstash-output-loki
  imageTag: 1.0.1

  logstashConfig:
    logstash.yml: |
      http.host: 0.0.0.0
      xpack.monitoring.enabled: false

  logstashPipeline:
    logstash.conf: |
      input {
        beats {
          port => 5044
        }
      }

      filter {
        if [kubernetes] {
          mutate {
            add_field => {
              "container" => "%{[kubernetes][container][name]}"
              "namespace" => "%{[kubernetes][namespace]}"
              "pod" => "%{[kubernetes][pod][name]}"
            }
            replace => { "host" => "%{[kubernetes][node][name]}"}
          }
        }
        mutate {
          remove_field => ["tags"]
        }
      }

      output {
        loki {
          url => "http://loki:3100/loki/api/v1/push"
        }
        # stdout { codec => rubydebug }
      }
```

### PR DESCRIPTION
**Pull request by Paszymaja**
Exemplary values in Loki-stack/values.yaml use probably some old version of logstash subchart and doesn't work.
Fixes #136

### CODE REVIEW

**Comment by AlexsJones** [comment]
Thank you for this, it's an annoying issue I faced also.

File: charts/loki-stack/values.yaml [hunk 1]
```diff
@@ -34,19 +34,31 @@ filebeat:
             - logs_path:
                 logs_path: "/var/log/containers/"
       output.logstash:
-        hosts: ["logstash-loki:5044"]
+        hosts: ["loki-logstash-headless:5044"] # {{name_of_release}}-logstash-headless:5044
```
Comment (skorobogatydmitry) [inline charts/loki-stack/values.yaml:37]: Why not 
```
  service:
    ports:
      - name: "5044"
        port: 5044
        targetPort: 5044
```
on logstash side ?

Comment (wsszh) [reply charts/loki-stack/values.yaml:37]: I think it is worth to merge the commits. As I post [here](https://github.com/grafana/helm-charts/issues/175), I am not the only one that can't use the chart to use Loki with filebeat without changing the `output.logstash.hosts` from `["logstash-loki:5044"]` to `["logstash-loki-headless:5044"]`. I think it is not cool for the chart if we can't use it directly?

**Review by wsszh — APPROVED** [review]

File: charts/loki-stack/values.yaml (hunk 1 above)
Comment (wsszh) [reply charts/loki-stack/values.yaml:37]: > Why not
> 
> ```
>   service:
>     ports:
>       - name: "5044"
>         port: 5044
>         targetPort: 5044
> ```
> 
> on logstash side ?

Hey, man! Although all the pods is running after I changed the `output.logstash.hosts` from `["logstash-loki:5044"]` to `["logstash-loki-headless:5044"]`, I can not get any labels on the Grafana website. After I edited the values.yaml as you post, then everything works! I don't know why...

**Comment by trevorwhitney** [comment]
Closing as the PR is rather old. Please open a new PR if still relevant, thanks.

**Pull request closed without merging**